### PR TITLE
Eigen includes are needed in more places, etc.

### PR DESCRIPTION
**Pull request by rpavlik**
Also remove erroneous redundant extlib/eigen mentions, and fix lintian warnings about the desktop files, etc.

Found when packaging rc1 for debian.

### CODE REVIEW

File: res/freedesktop/solvespace-snap.desktop [hunk 1]
```diff
@@ -3,7 +3,7 @@ Version=1.0
 Name=SolveSpace
 Comment=A parametric 2d/3d CAD
 Exec=solvespace
-MimeType=application/x-solvespace
+MimeType=application/x-solvespace %f
```
Comment (ktrace) [inline res/freedesktop/solvespace-snap.desktop:6]: I don't think that mimetype need parameter, look like it need in line 5

Comment (rpavlik) [reply res/freedesktop/solvespace-snap.desktop:6]: Gah good catch

Comment (rpavlik) [reply res/freedesktop/solvespace-snap.desktop]: Fixed!

**Comment by rpavlik** [comment]
OK, fixed my goof from earlier

**Comment by phkahler** [comment]
@rpavlik  you can merge this when you're happy with it.

**Pull request merged**